tensorflow/core/kernels/cwise_op_gpu_select.cu.cc

```diff
@@ -33,6 +33,20 @@ struct SelectFunctor<GPUDevice, T> {
 };
 
 template <typename T>
+struct SelectScalarFunctor<GPUDevice, T> {
+  void operator()(const GPUDevice& d, typename TTypes<T>::Flat out,
+                  typename TTypes<bool>::ConstScalar cond,
+                  typename TTypes<T>::ConstFlat then_flat,
+                  typename TTypes<T>::ConstFlat else_flat) {
+    if (cond()){
```

Comment:
You can put cond on the host by marking that tensor to be in HostMemory during the GPU kernel registration.

I think for the assignments, you don't need to To32Bit(out)...

just do:

out.device(d) = then_flat;

etc.